### ISSUE TITLE
Add support for pluggable downloaders

### ISSUE THREAD

**Issue by sabbyanandan** [spring-cloud-stream-app-starters/sftp#24]
Currently the Sftp source Task Launch Request supports the file-ingest use case but requires the launched task to also connect to the sftp server to read the referenced file. If running in the cloud, it is preferable to download to a shared persistent staging area, such as an NFS mount provided by PCF Volume Services, (or a K8s PVC).  In this scenario, Spring Batch restart capabilities would be supported in the event that the task container crashed before completing ingest of a large file.  

Given this premise, it is more convenient for the sftp source to provide the option to stage the file to an interim store.  This is true even for a file ingest task running locally.  

The sftp source should implement the spring-cloud-stream function support provided in the scst 2.1 release. Then we can provide a standard library of Function beans (using spring-cloud-function) to implement downloaders for local, nfs, and s3.

**Comment by dturanski** [comment]
This may be considered `enable spring-cloud-stream function support`.  There are some specific message headers added in task launch request messages to support downloading "local" or remote files.  This is implemented in https://github.com/dturanski/sftp/tree/stream-function-support . A PR will depend on the Einstein release train.

**Comment by dturanski** [comment]
Also the Downloader implementations are https://github.com/dturanski/sftp-downloaders

**Comment by dturanski** [comment]
TLDR; this no longer requires scst function support.  Following review with @markpollack and @sabbyanandan, . We decided that the downloaders should be included as dependencies for the source to not require users to pull and build the app to enable this feature. Given this, the current direction is not to use spring-cloud-stream function support for this in favor of a regular service activator. The main reason for this is that the downloader is auto configured via the property `sftp.transfer-to={LOCAL,S3,CF_VOLUME}` . If this also requires a declared function definition, things get unnecessarily complex.

**Comment by dturanski** [comment]
I labeled `invalid` and will close this since it is obsolete.  The sftp source functionality will remain as is. The task launch request only works with remote files and will remain as `legacy`. A new `sftp-dataflow` source will handle local file access.
